openpype/hosts/maya/plugins/publish/validate_resolution.py

```diff
@@ -0,0 +1,81 @@
+import pyblish.api
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin
+)
+from maya import cmds
+from openpype.pipeline.publish import RepairAction
+from openpype.hosts.maya.api import lib
+from openpype.hosts.maya.api.lib import reset_scene_resolution
+
+
+class ValidateSceneResolution(pyblish.api.InstancePlugin,
+                              OptionalPyblishPluginMixin):
+    """Validate the scene resolution setting aligned with DB"""
+
+    order = pyblish.api.ValidatorOrder - 0.01
+    families = ["renderlayer"]
+    hosts = ["maya"]
+    label = "Validate Resolution"
+    actions = [RepairAction]
+    optional = True
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+        width, height, pixelAspect = self.get_db_resolution(instance)
+        current_renderer = cmds.getAttr(
+            "defaultRenderGlobals.currentRenderer")
```

Comment:
This one should also be acquired from the renderlayer, but you're in luck - it already is collected. I believe it's `instance.data["renderer"]`.